### ISSUE TITLE
Need a System to Skip Harvesting Harvested Packages

### ISSUE THREAD

**Issue by jayhuggins** [ckan/ckanext-harvest#129]
We're running into issues when CKAN A harvests from CKAN B, and CKAN C harvests from CKAN A and B.  It would be ideal to have a configuration setting to skip harvested packages (which should probably default to True).

**Comment by davidread** [comment]
The problem is not avoiding harvested datasets, the problem is avoiding duplicates. Then if CKAN B starts harvesting from a number of other places then you'll want them harvested into both CKAN A and C too, by default.

I think that the solution is to record in a dataset some sort of 'history' of it being harvested, and record a unique ID. Remember that you can harvest from lots of other types of data stores aside from CKANs and they record IDs in different ways, so we need to record the source's ID. I've been suggesting that when you harvest a record you store/append to the 'provenance' extra which contains the info you need to know where the record came from and avoid reharvesting if you already harvested it. The tech details are here:

https://gist.github.com/davidread/979a7d63f6d813052255

**Comment by jayhuggins** [comment]
Correct, the issue is avoiding duplicates by avoiding harvested datasets.  Thought if B starts harvesting from other places, we would prefer not to pick these up by default.  If we want to harvest from these other places, we'll add them as another harvest source.  Others may feel differently and to accommodate both, it would be nice to have a configuration option.

To clarify the way this played out, everything worked fine initially with the topmost harvest process correctly identifying duplicates and skipping them.  In subsequent reharvests that were not performed in the same sequence, we ended up with two harvest sources owning packages that should only have been owned by one, and attempts to clear the harvest source resulted in crashes from database constraint violations (specifically in the sql block in harvest_source_clear, ckanext-harvest/ckanext/harvest/logic/action/update.py).
